frontend/crd/src/app/milestones-page/milestones/milestones.component.ts

```diff
@@ -24,14 +31,41 @@ export class MilestonesComponent implements OnInit, OnDestroy {
 
   ngOnInit() {
     this.milestoneService.getMilestones()
-      .pipe(takeUntil(this.destroyed$))
-      .subscribe((milestones: Milestone[]) => {
-        this.yearLevels.forEach((yearLevel) => this.milestonesMap.set(yearLevel, new Array<Milestone>()));
-        milestones.forEach((milestone) => this.milestonesMap.get(milestone.yearLevel)?.push(milestone));
+      .pipe(takeUntil(this.destroyed$),
+        switchMap((milestones: Milestone[]) => {
+          this.yearLevels.forEach((yearLevel) => this.milestonesMap.set(yearLevel, new Array<Milestone>()));
+          milestones.forEach((milestone) => this.milestonesMap.get(milestone.yearLevel)?.push(milestone));
+          
+          return this.authService.user$;
```

Comment:
separate the milestone stuff from the submission stuff

Reply:
what do you mean?